### ISSUE TITLE
clone operation failed with access denied error

### ISSUE THREAD

**Issue by rickysarraf** [systemd/systemd#2867]
### Submission type
- [x] Bug report
- [ ] Request for enhancement (RFE)

_NOTE: Do not submit anything other than bug reports or RFEs via the issue tracker!_
### systemd version the issue has been seen with

> 229

_NOTE: Do not submit bug reports about anything but the two most recently released systemd versions upstream!_
### Used distribution

> Debian Unstable
### In case of bug report: Expected behaviour you didn't see

> clone operation should have succeeded.
### In case of bug report: Unexpected behaviour you saw

> Got error message
### In case of bug report: Steps to reproduce the problem

```
rrs@learner:~/.rrs-home/Community/FlameGraph (master)$ machinectl list-images 
NAME               TYPE      RO  USAGE  CREATED                     MODIFIED
debTemplate        subvolume no  849.6M Mon 2015-09-21 20:34:50 IST n/a     
fedora22           subvolume no  1.2G   Tue 2016-03-15 22:36:20 IST n/a     
jessie             directory no  n/a    n/a                         n/a     
rawhideFedora      subvolume no  1.5G   Mon 2015-09-21 20:36:40 IST n/a     
sanboot            subvolume no  1.2G   Fri 2016-03-04 16:19:18 IST n/a     
trustyTemplate     subvolume no  414.6M Mon 2015-09-21 20:37:15 IST n/a     
tumbleweedTemplate subvolume no  1.2G   Mon 2015-09-21 20:37:37 IST n/a     

7 images listed.
2016-03-18 / 22:20:12 ♒♒♒  ☺  
rrs@learner:~/.rrs-home/Community/FlameGraph (master)$ machinectl clone fedora22 test
2016-03-18 / 22:20:24 ♒♒♒  ☺  
rrs@learner:~/.rrs-home/Community/FlameGraph (master)$ machinectl clone jessie temp
Could not clone image: Access denied
2016-03-18 / 22:20:39 ♒♒♒  ☹  => 1  
```

I noticed that Issue #465 was opened some time ago. But was later closed as no more reproducible. I've also put my comments and logs there.

**Comment by poettering** [comment]
Does it work if you are root? If so, then this appears as if we have a bug in our PolicyKit policy!

**Comment by rickysarraf** [comment]
Nope. The error msg remains the same.

```
rrs@learner:~$ machinectl list-images 
NAME               TYPE      RO  USAGE  CREATED                     MODIFIED
debTemplate        subvolume no  849.6M Mon 2015-09-21 20:34:50 IST n/a     
fedora22           subvolume no  1.2G   Tue 2016-03-15 22:36:20 IST n/a     
jessie             directory no  n/a    n/a                         n/a     
rawhideFedora      subvolume no  1.5G   Mon 2015-09-21 20:36:40 IST n/a     
trustyTemplate     subvolume no  414.6M Mon 2015-09-21 20:37:15 IST n/a     
tumbleweedTemplate subvolume no  1.2G   Mon 2015-09-21 20:37:37 IST n/a     

6 images listed.
2016-04-15 / 17:12:49 ♒♒♒  ☺  

rrs@learner:~$ sudo machinectl clone jessie test
[sudo] password for rrs: 
Could not clone image: Access denied
2016-04-15 / 17:13:02 ♒♒♒  ☹  => 1  
```

**Comment by rickysarraf** [comment]
And in case you need some info about the directory based container.

```
rrs@learner:~$ machinectl status jessie
jessie
           Since: Fri 2016-04-15 17:22:37 IST; 30s ago
          Leader: 13573 (systemd)
         Service: systemd-nspawn; class container
            Root: /var/lib/machines/jessie
           Iface: sysbr0
         Address: 172.16.15.159
                  fe80::4c6b:17ff:fec6:6111%3
              OS: Debian GNU/Linux 8 (jessie)
            Unit: systemd-nspawn@jessie.service
                  ├─13570 /usr/bin/systemd-nspawn --quiet --keep-unit --boot --link-journal=try-guest --network-bridge=sysbr0 --bind /var/tmp/lxc --settings=override --machine=jessie
                  ├─13573 /lib/systemd/systemd
                  └─system.slice
                    ├─dbus.service
                    │ └─13677 /usr/bin/dbus-daemon --system --address=systemd: --nofork --nopidfile --systemd-activation
                    ├─cron.service
                    │ └─13674 /usr/sbin/cron -f
                    ├─networking.service
                    │ └─13655 dhclient -v -pf /run/dhclient.host0.pid -lf /var/lib/dhcp/dhclient.host0.leases host0
                    ├─systemd-journald.service
                    │ └─13605 /lib/systemd/systemd-journald
                    ├─systemd-logind.service
                    │ └─13676 /lib/systemd/systemd-logind
                    └─console-getty.service
                      └─13682 /sbin/agetty --noclear --keep-baud console 115200 38400 9600 vt102

Apr 15 17:22:37 learner systemd-nspawn[13570]: [  OK  ] Started Console Getty.
Apr 15 17:22:37 learner systemd-nspawn[13570]: [  OK  ] Reached target Login Prompts.
Apr 15 17:22:37 learner systemd-nspawn[13570]: [  OK  ] Reached target Multi-User System.
Apr 15 17:22:37 learner systemd-nspawn[13570]: [  OK  ] Reached target Graphical Interface.
Apr 15 17:22:37 learner systemd-nspawn[13570]:          Starting Update UTMP about System Runlevel Changes...
Apr 15 17:22:37 learner systemd-nspawn[13570]: [  OK  ] Started Cleanup of Temporary Directories.
Apr 15 17:22:37 learner systemd-nspawn[13570]: [  OK  ] Started Update UTMP about System Runlevel Changes.
Apr 15 17:22:38 learner systemd-nspawn[13570]: [2B blob data]
Apr 15 17:22:38 learner systemd-nspawn[13570]: Debian GNU/Linux 8 learner console
Apr 15 17:22:38 learner systemd-nspawn[13570]: [1B blob data]
2016-04-15 / 17:23:08 ♒♒♒  ☺  
```

**Comment by poettering** [comment]
@rickysarraf any chance you can get me the file "strace -f -o log -p $PID" generates (with $PID replaced by machine'ds PID), when you issue the "machinectl clone" operation that fails? Invoke the strace immediately before, run the clone from another window, then abort the strace with ctrl-c. It generates a file called "log", please attach that file here (pasting hte last 100lines or so if it is fine, if it is overly large)

**Comment by poettering** [comment]
@rickysarraf oh, I see now that you already appended one to #465. And that log is great! It tells me exactly what is going wrong. Will prep fix shortly.

**Comment by poettering** [comment]
Fix waiting in #3116.
